### ISSUE TITLE
unyt array __format__ prints units twice

### ISSUE THREAD

**Issue by umesh-timalsina** [yt-project/unyt#183]
* unyt version: `v2.8.0`
* Python version: `3.7.9`
* numpy version : `1.19.4`
* Operating System: Linux Mint Tricia

### Description
It looks like `__format__` dunder is unimplemented for numpy arrays. That's why it passed to the `__str__` method and using `unyt_array` in format strings prints units twice.
### What did I do?

```
>>> import unyt as u
>>> import numpy as np
>>> arr = np.array([1, 1., 1]) * u.nm
>>> f"{arr}"
'[1. 1. 1.] nm nm'
>>> 

```

**Comment by brittonsmith** [comment]
This may be fixed by PR #188.

**Comment by neutrinoceros** [comment]
thanks for linking @brittonsmith, I forgot to check the issue tracker !
